### PR TITLE
sarahpeng/phs 35 email collection frontend

### PR DESCRIPTION
**Pull request by sarahhpeng**
# ✨ New in this PR ✨

## One liner: what did you do? 🛠
Created the email pop up

## Coverage 🙆‍♀️

- Created a pop up that is open by default, but can be closed if you click the x 
- Then added all the components/styling/routing/conditional rendering/icons
- If you just click subscribe with nothing in the input, it displays an error message, and if the email is not valid(aka does not have an @ and does not end with either .com or .ca), another error message appears
- If the email is 'valid,' once the user clicks subscribe, it leads the user to another front end of the pop up with a back to home button, that the user can click, and go to the home page! (but right now, it's the collections page)

## How can the reviewer test your code? Where is this sprint located in the codebase? 👩‍💻
- localhost:3000/dummyPage

## REQUIRED: Link a short video demo of the feature you have added. 👩‍💻
https://www.loom.com/share/fabcf5c80966467b8d38a514b448d3fd?sid=95bb9da6-a1f5-43e4-9950-610a998a05fc

## Any bugs you encountered or still having trouble with? 🐛
- Need to overlap on top of home page and the back to home button has to go to the home screen, but I don't know what that's called, so I just routed it to the collections page

## Resources 📔
Andrei

_REQUIREMENTS in this section: Link all relevant online resources and related PRs. Write down any new dependencies or packages and link to their documentation. If you ran into bugs, link any stack overflow or online resources that helped you unblock yourself_

## ❗IF this sprint was marked as FRONTEND on linear, cc Claire Wang

_REMEMBER: once PR is made, also request a PR review Andrei, CC'ing is NOT enough!_
🏚 cc: @andreisito @cluurie

### CODE REVIEW

**Review by andreisito — COMMENTED** [review]
rename your file to fit PascalCase instead of camelCase

File: src/components/userComponents/emailPopup/page.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+import { Dialog, Transition } from '@headlessui/react';
+import { PiPaperPlaneTiltBold , PiSealCheck } from "react-icons/pi";
+import { VscClose } from "react-icons/vsc";
+import { BiErrorCircle } from "react-icons/bi";
+
+import Link from 'next/link';
+
+
+/**
+ * @returns an email pop up
```
Comment (andreisito) [inline src/components/userComponents/emailPopup/page.tsx]: add docs for the error case and conditional rendering

File: src/components/userComponents/emailPopup/page.tsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+                      <div className="flex items-center justify-center pt-[24px]">
+                      <button
+                        type="button"
+                        className="bg-[#7CA24E] w-[283px] h-[43px] text-white rounded-2xl p-[21px] mt-[8px] flex items-center justify-center"
+                      >
+                        Back to Home
+                      </button>
+                    </div>
+                    </Link>
```
Comment (andreisito) [inline src/components/userComponents/emailPopup/page.tsx]: For readability/maintainability:  Lets define this success page as its own helper component somewhere earlier in this file

File: src/components/userComponents/emailPopup/page.tsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                      onChange={handleChange}
+                      className={`font-Lato text-sm bg-gray-100 border rounded-md pl-[13px] w-[283px] h-[43px] items-center ${
+                        inputValue ? 'text-black' : 'text-[#BDBDBD]'
+                      }`}
+                    />
+                  </div>
+
+                  <div className="flex items-center justify-center">
+                    <button
+                      type="button"
```
Comment (andreisito) [inline src/components/userComponents/emailPopup/page.tsx]: readability: lets also move this to be a helper component that we call in this segment.  Will be much easier to maintain this way.  I would also recommend switching the rendering cases here -> This should be the first thing that renders based on if !subscribed instead.

File: src/components/userComponents/emailPopup/page.tsx [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+                    </Link>
+
+                      </>
+                    ) : (
+                      <>
+
+                  <div className="absolute top-[102px] left-1/2 transform -translate-x-1/2 -translate-y-1/2">
+                  <PiPaperPlaneTiltBold className="text-[#3F6A38] text-4xl" />
+                  </div>
+
```
Comment (andreisito) [inline src/components/userComponents/emailPopup/page.tsx]: styling:  move to using tailwind units on this whole file.  You can ask chatgpt to make hte conversion or use this website: https://tailwindremconverter.netlify.app/

File: src/components/userComponents/emailPopup/page.tsx [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+import { VscClose } from "react-icons/vsc";
+import { BiErrorCircle } from "react-icons/bi";
+
+import Link from 'next/link';
+
+
+/**
+ * @returns an email pop up
+ */
+export default function EmailPopup(
```
Comment (andreisito) [inline src/components/userComponents/emailPopup/page.tsx]: feat:  After NPO feedback, we are going to be using EmailPopup in more places.  Lets add a parameter called "backLink" or something.  We can use this to define where a popup redirects.  Ask/text me for details if confused, I'll add another comment below of what I'm talking about.

File: src/components/userComponents/emailPopup/page.tsx [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+                        THANKS FOR SUBSCRIBING!
+                      </h2>
+                      
+                      <p className="text-center text-sm font-Lato text-[#3B3B3B] pt-[10px] pl-[30px] pr-[30px] text-[16px]">
+                      Your sign-up request was successful! Please check your email inbox to confirm.</p>
+                      
+
+                    <Link href="/collectionsPage" >
+
+                      <div className="flex items-center justify-center pt-[24px]">
```
Comment (andreisito) [inline src/components/userComponents/emailPopup/page.tsx]: feat:  We are going to refactor the backLink to be customizable

**Review by andreisito — CHANGES_REQUESTED** [review]
overall:  Move some div blocks into separate components for readability.  Convert px units to tailwind.  Refactor a bit of this to support parameters s.t. we can use this in multiple places on the app.

File: src/components/userComponents/emailPopup/page.tsx [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+  const handleChange = (e) => {
+    setInputValue(e.target.value);
+    setShowError(false);
+  };
+
+  const isValidEmail = (email) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.(com|ca)$/;
+    return emailRegex.test(email);
+  };
+
```
Comment (andreisito) [inline src/components/userComponents/emailPopup/page.tsx]: feat: so that we hit all potential email edge cases, could you add support for the following email suffixes.
.edu: Education institutions
.gov: Government entities
.org: Non-profit organizations
.net: Network-related organizations

Comment (miha-bhaskaran) [reply src/components/userComponents/emailPopup/page.tsx]: - make this more rigorous either by validating the text between @ and . as currently kladj;falksdf@a;dlfkjasd;.com is accepted
- look up email validation rules
- incorporate more extensive regex
- maybe potentially a email validation API?

**Review by miha-bhaskaran — DISMISSED** [review]
needed rigorous email validation

File: src/components/userComponents/emailPopup/page.tsx (hunk 4 above)
Comment (sarahhpeng) [reply src/components/userComponents/emailPopup/page.tsx]: tailwind conversions did not work, but will come back to this if it is a problem. just won't make this a habit.

**Review by andreisito — APPROVED** [review]
